server/routes/api/auth.ts

```diff
@@ -89,11 +112,17 @@ router.use('/users', passport.authenticate('jwt', {session: false}), requireAdmi
  */
 router.post<unknown, unknown, AuthAuthenticationOptions>('/authenticate', (req, res) => {
   if (config.disableUsersAndAuth) {
-    sendAuthenticationResponse(res, Users.getConfigUser());
+    const response = getAuthenticationResponse(res, Users.getConfigUser());
+    res.json(response);
     return;
   }
 
-  const parsedResult = authAuthenticationSchema.safeParse(req.body);
+  let parsedResult = authAuthenticationSchema.safeParse(null);
```

Comment:
Schema is not necessary as no complex data structure is involved in this case and incoming requests are already validated and authenticated by the web server (Nginx).

Simply parse the username from the header and do a null check.

Reply:
I have to pass to `Users.comparePassword` a `Credentials` type in the end and this way every new external authorization handler can return `authAuthenticationSchema`

since I'm not so proficient in ts I didn't find a better way to cast